### PR TITLE
Upgrade to Splunk 7.2 Cloud

### PR DESCRIPTION
**Pull request by kbalante**
A. Remove all references that save the api key to the file system in plain text and have it save and retrieve from the encrypted storage/passwords rest service instead. 
- this was part of the splunk-appinspect tool findings

B. Also remove the logging of sensitive data in the url to the log files
- this was part of the splunk-appinspect tool findings

C. Cleaned up appserver/static/js/views/SetupView.js
- there was a lot of unused copy and paste code in SetupView.js from a gist online somewhere
- some of this code wasn't working and wasn't referenced anywhere so I removed it to prevent confusion

D. On page load, remove existing plain text api key (if it's there) and copy into an encrypted format using the storage/passwords splunk service
- this is required to migrate existing customers to the new app version

E. Upgrade documentation added in the CHANGELOG

### CODE REVIEW

File: appserver/static/js/views/SetupView.js [hunk 1]
```diff
@@ -99,21 +99,6 @@ define([
 	    }
 	});
 
-    var EncryptedCredential = SplunkDBaseModel.extend({
```
Comment (kbalante) [inline appserver/static/js/views/SetupView.js:102]: cleanup of unused code in a bunch of places in this file

File: bin/util/stream_consumer.py [hunk 2]
```diff
@@ -25,7 +25,6 @@ def __init__(self, connection_data, on_event_callback):
 
     def run(self):
         try:
-            logger.debug('Connecting to {}'.format(self._url))
```
Comment (kbalante) [inline bin/util/stream_consumer.py:28]: this is where it was logging confidential info

File: appserver/static/js/views/Amp4eEventsInputListView.js [hunk 3]
```diff
@@ -95,9 +98,22 @@ define([
                 url: '/splunkd/__raw/services/configs/conf-inputs/amp4e_events_input',
                 success: function (model, _response, _options) {
                     console.info("Successfully retrieved the default amp4e_events_input configuration");
+
+                    // migrate to new secure format for api key if an api key exists in unsecure format
```
Comment (kbalante) [inline appserver/static/js/views/Amp4eEventsInputListView.js:102]: migration from plain text to encrypted api key

**Review by pschulze — DISMISSED** [review]
Looks great @kbalante, thanks for getting this done!

**Comment by kbalante** [comment]
using this instead https://github.com/Cisco-AMP/amp4e_splunk_events_input/pull/37

**Pull request closed without merging**